### PR TITLE
Parameterless constructor for TypeMapConverter

### PR DESCRIPTION
**Pull request by jenniferplusplus**
This allows the JsonLdSerializer to be used as an AspNet formatter.

### CODE REVIEW

**Comment by jenniferplusplus** [comment]
@warriordog have you had a chance to review this? I'll need something like this in order to make APSharp.Types types usable as args or return values in ASP.NET core controllers.

File: Source/ActivityPub.Types/Conversion/ASTypeInfoCache.cs
Comment (warriordog) [inline Source/ActivityPub.Types/Conversion/ASTypeInfoCache.cs:1]: tbh, I'm really uncomfortable with making this a static singleton *and* publicly exposed. What if we made a separate `StaticASTypeInfoCache` instead? That could hold the public shared instance, which would still be of type `ASTypeInfoCache`.

File: Source/ActivityPub.Types/Conversion/ASTypeInfoCache.cs [hunk 1]
```diff
@@ -59,6 +59,12 @@ public interface IASTypeInfoCache
 /// </summary>
 public class ASTypeInfoCache : IASTypeInfoCache
 {
+    private static readonly Lazy<ASTypeInfoCache> Lazy = new(() =>
+    {
+        var cache = new ASTypeInfoCache();
+        cache.RegisterAllAssemblies();
```
Comment (warriordog) [inline Source/ActivityPub.Types/Conversion/ASTypeInfoCache.cs:65]: I've thought about this a lot, and I think it makes more sense to move this call to the constructor. Then we can remove the two public "Register" methods, which makes this class immutable.

**Review by warriordog — APPROVED** [review]
@jenniferplusplus approved, but I left a suggestion for a slightly different approach that might be better. I'll leave the final decision up to you, since I don't want to hold back your work any longer.

**Comment by warriordog** [comment]
@jenniferplusplus Actually, I'm going to go ahead and merge this. Then I can go ahead and start work in #153, which will get rid of this class entirely.

**Comment by jenniferplusplus** [comment]
@warriordog cool, thanks. Don't worry, these issues aren't significant blockers for me. I have no trouble making local changes to ActivityPubSharp for experiments. In the (hopefully, very) near future I'll need published packages to depend on, for a merge back to main. But that hasn't been a problem so far.

And for what it's worth, the thing I tried worked. I was trying a more proper use of Formatters, which do have access to DI, and that worked. It's not even as annoying to configure as I had been telling myself it would be.

Ex:
https://github.com/Letterbook/Letterbook/blob/6596d3ee34552b84c6fd4d0f8e61c21f65cb92ef/Letterbook.Adapter.ActivityPub/JsonLdInputFormatter.cs#L22-L33

(This route might make it harder to depend on derived types as input to controllers, but, I'm not even sure that's a real problem.)

**Comment by warriordog** [comment]
@jenniferplusplus does that mean that you don't need the static instance now?

**Comment by jenniferplusplus** [comment]
@warriordog I believe so, yes. I'm like a month+ into this very messy trial-and-error initial interop work, and it's a little hard to be sure I'm not depending on something I only have locally. But I really (really) hope that's wrapping up the next day or two, and it will be a lot easier to confirm then.

**Comment by warriordog** [comment]
@jenniferplusplus ok, please let me know when you have an answer! I'd love to remove the static stuff if possible.

**Comment by jenniferplusplus** [comment]
@warriordog Just letting you know I haven't forgotten about this. I'll get to it soon, but probably not before christmas.

**Pull request merged**